dsc/docs-conceptual/dsc-2.0/tutorials/create-dsc-resource-machine-config.md

```diff
@@ -0,0 +1,2086 @@
+  - The `User` configuration file is `~/Library/Preferences/TailSpinToys/tstoy/tstoy.config.json`
+
+To handle these paths, you need to create a helper method, `GetConfigurationFile()`.
+
+```powershell
+[string] GetConfigurationFile() {
+    $FilePaths = @{
+        Linux = @{
+            Machine   = '/etc/xdg/TailSpinToys/tstoy/tstoy.config.json'
+            User      = '~/.config/TailSpinToys/tstoy/tstoy.config.json'
+        }
+        MacOS = @{
+            Machine   = '/Library/Preferences/TailSpinToys/tstoy/tstoy.config.json'
+            User      = '~/Library/Preferences/TailSpinToys/tstoy/tstoy.config.json'
+        }
+        Windows = @{
+            Machine = "$env:ProgramData\TailSpinToys\tstoy\tstoy.config.json"
+            User    = "$env:APPDATA\TailSpinToys\tstoy\tstoy.config.json"
+        }
+    }
+
+    $Scope = $this.ConfigurationScope.ToString()
+
+    if ($Global:PSVersionTable.PSVersion.Major -lt 6 -or $Global:IsWindows) {
+        return $FilePaths.Windows.$Scope
+    } elseif ($Global:IsLinux) {
+        return $FilePaths.Linux.$Scope
+    } else {
+        return $FilePaths.MacOS.$Scope
+    }
+}
+```
+
+To test this new method, execute the `using` statement to load the **ExampleResources** module's
+classes and enums into your current session.
+
+```powershell
+using module ./ExampleResources.psd1
+$Example = [Tailspin]::new()
+$Example
+$Example.GetConfigurationFile()
+$Example.ConfigurationScope = 'User'
+$Example.GetConfigurationFile()
+```
+
+```Output
+Ensure  ConfigurationScope UpdateAutomatically UpdateFrequency
+------- ------------------ ------------------- ---------------
+Present            Machine               False               0
+
+C:\ProgramData\TailSpinToys\tstoy\tstoy.config.json
+
+C:\Users\mikey\AppData\Roaming\TailSpinToys\tstoy\tstoy.config.json
+```
+
+Open `Helpers.ps1` in VS Code. Copy and paste the paths for the configuration files into the script,
+assigning them to `$TSToyMachinePath` and `$TSToyUserPath`. The file should look like this:
+
+```powershell
+$env:PSModulePath += "<separator>$pwd"
+$TSToyMachinePath = '<machine configuration file path>'
+$TSToyUserPath = '<user configuration file path>'
+```
+
+Exit the terminal in VS Code and open a new terminal. Dot-source `Helpers.ps1`.
+
+```powershell
+. ./Helpers.ps1
+```
+
+Now you can write the rest of the `Get()` method.
+
+```powershell
+[Tailspin] Get() {
+    $CurrentState = [Tailspin]::new()
+
+    $CurrentState.ConfigurationScope = $this.ConfigurationScope
+
+    $FilePath = $this.GetConfigurationFile()
+
+    if (!(Test-Path -Path $FilePath)) {
+        $CurrentState.Ensure = [TailspinEnsure]::Absent
+
+        $this.CachedCurrentState = $CurrentState
+
+        return $CurrentState
+    }
+
+    $Data = Get-Content -Raw -Path $FilePath |
+        ConvertFrom-Json -ErrorAction Stop
+
+    $this.CachedData = $Data
+
+    if ($null -ne $Data.Updates.Automatic) {
```

Comment:
I've had mixed results when evaluating strings against null. Your call.
```suggestion
    if (![string]::IsNullOrEmpty($Data.Updates.Automatic)) {
```

Reply:
This is a check against boolean true/false - the value might not exist in the returned data (as converted from JSON).